apps/antalmanac/src/actions/AppStoreActions.ts

```diff
@@ -12,6 +12,11 @@ import AppStore from '$stores/AppStore';
 import trpc from '$lib/api/trpc';
 import { courseNumAsDecimal } from '$lib/analytics';
 
+export interface CallbackOptions {
+    onSuccess: (name?: string) => unknown;
+    onError: (name?: string) => unknown;
```

Comment:
The function signature of `copySchedule` says that the `name` is required, so we might as well guarantee that the name will be passed to the callbacks as well.

Also, could you pass the index? I don't think there's any need to modify the function signature in this way for this particular case.

```suggestion
    onSuccess: (index: number) => unknown;
    onError: (index: number) => unknown;
```